### ISSUE TITLE
Improve PFT classifier

### ISSUE THREAD

**Issue by annaspiers** [RS-PRISMATIC/PRISMATIC#19]
### Problem
There are a number of ways we can improve our existing plant functional type (PFT) classifier.

### Details
See to do list below for exhaustive list of enhancements. 

### To do
- [ ] Read lit
  - [ ] See Fabian and Marcos’ ED2 paper for reference (Fabian)
  - [ ] See Fabian’s work: PFT classification for Lägern (ED2 paper). Continuum removal approach to normalise the spectrum and absorption features  across the bands
  - [ ] Do some basic reading on species/PFT classification in lidar and his imagery
    - [ ] How do others generate training data?  Common strategies for shrub, bare ground, grass training for PFT classification?
- [ ] Automate using hytools-corrected flightlines instead of NEON tiles
  - [ ] Use already corrected hyperspectral flightiness. crop to the extent of the other AOP tiles used in Scholl workflow. Save in project hyper spectral folder - script 04
  - [ ] Am I applying glint correction in hytools?
- [ ] Training data
  - [ ] Generate bareground/rock/grass/sand by using CHM and NDVI masks
    - [ ] Then change chm>0 line in prep_features_for_RF
  - [ ] Automate incorporation of manually delineated crowns (mine and Carissa's)
  - [ ] An alternative way to create training data is to 1) use max crown diameter, 2) keep pixels in that polygon that are within say ‘75%’ of max canopy height (from CHM) in that polygon. This way we feel confident that the pixels kept for that polygon and pft are from that individual, and not from another individual or the ground
  - [ ] Look into differences in PFT distribution between RS and inventory data
    - [ ] Marcos: This is something that may be worth taking a closer look. It can well be that the plots are in more open areas, whereas the entire domain is more dominated by cedar forests. 
    - [ ] Anna: It could also be a product of cedars being more visible to the imaging spectrometer even if there are a bunch of shrubs or oaks in the understory. Or maybe the classification algorithm has a bias toward cedars. 
- [ ] Data preparation
  - [ ] Make sure pixels from the same individual are either in train OR validation (Fabian)
  - [ ] Continuum removal? Try it out - explore  (Fabian)
  - [ ] Switch lupinus to shrub (Fabian)
  - [ ] Break ‘other’ into subcategories like other conifer, other broadleaves (Fabian)
  - [ ] Balance the training data by category
  - [ ] Add spectral indices and nbr as predictors
  - [ ] Calculate number of PCs in an informed way. Look into scree plot (“elbow approach”). Fabian used top 5 or 6 PCs (less than 100 but more than 3) 
    - [ ] may be good to exclude PC1 because it’s mostly brightness/shade (Fabian)
  - [ ] Apply shade mask - Crown shadows are challenging (Fabian)
  - [ ] Try vector normalization before PC (Fabian)
  - [ ] Try shadow masking (mask out shadow, - do this in stacked-aop for training and test imagery) (Fabian)
- [ ] Model
  - [ ] upgrade from RF to XGboost?
  - [ ] Think through changing workflow from predicting PFT per pixel to per shape
  - [ ] train classifier on all site-years at once (not individually) - first bind the mega-cv’s, then fit classifier 
  - [ ] Shashi has python code for svm and neural network he could share if I’d like
- [ ] Model validation
  - [ ] Try plotting these PCs as an RGB composite to get an idea of what each component represents (Fabian)
  - [ ] Visualize the predicted map (Fabian)
  - [ ] visualize reference data and classification results
